### PR TITLE
Add Pylint check

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

Add Pylint check:
* Considered both Pylint and Ruff. Decided to go with Pylint to match G3 style, as we will need to have a sub-repo for NPI workflow.
* Use google style [here](https://github.com/google/styleguide/blob/gh-pages/pylintrc), but I will need to update `max-line-length` to 125 instead of 80 to match Pyink format style.
* I addressed some straightforward errors, will need following PRs to address rest.
* Currently, set the threshold to 9/10 to gradually increase our code style quality (currently dags folder is 10/10, and xlml folder is 9.2/10).
* Add github action, and update doc and scripts accordingly.

# Tests

Please describe the tests that you ran on Cloud VM to verify changes.

**Instruction and/or command lines to reproduce your tests:** ...
Test it locally and on PR checks.

**List links for your tests (use go/shortn-gen for any internal link):** ...
All checks passed.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: xlml/apis/test_config.py [hunk 1]
```diff
@@ -135,7 +135,6 @@ class TestConfig(abc.ABC, Generic[A]):
   @abc.abstractmethod
   def benchmark_id(self) -> str:
     """Unique key for metrics generated by this test."""
-    ...
```
Comment (will-cromar) [inline xlml/apis/test_config.py:138]: I assume pylint was complaining about an unused expression. Should this be `pass` or `raise NotImplementedError()` instead?

File: .github/workflows/pylint-check.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint
+    - name: Analysing the code with pylint
+      run: |
+        pylint ./dags --rcfile=scripts/.pylintrc --fail-under=9
```
Comment (will-cromar) [inline .github/workflows/pylint-check.yml:29]: Can you run `scripts/code-style.sh` here instead and check for diffs? I'd also like to add JSonnet styling there too (#149)

**Review by will-cromar — COMMENTED** [review]
> Use google style [here](https://github.com/google/styleguide/blob/gh-pages/pylintrc), but I will need to update max-line-length to 125 instead of 80 to match Pyink format style.

Can you change `pyink` to match the style guide instead?

File: xlml/apis/test_config.py (hunk 1 above)
Comment (RissyRan) [reply xlml/apis/test_config.py:138]: I will prefer `raise NotImplementedError()` here, especially for benchmark_id. Will update it accordingly.

**Comment by RissyRan** [comment]
> > Use google style [here](https://github.com/google/styleguide/blob/gh-pages/pylintrc), but I will need to update max-line-length to 125 instead of 80 to match Pyink format style.
> 
> Can you change `pyink` to match the style guide instead?

Can I update that to 80 in coming PR? It will be lots of changes there.

File: .github/workflows/pylint-check.yml (hunk 2 above)
Comment (RissyRan) [reply .github/workflows/pylint-check.yml:29]: Sure thing.

**Review by will-cromar — APPROVED** [review]

File: .github/workflows/pylint-check.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint
```
Comment (will-cromar) [inline .github/workflows/pylint-check.yml:26]: One more nit: can you lock this to a specific pylint version (at least a major version) so the check doesn't change when pylint updates?

Comment (RissyRan) [reply .github/workflows/pylint-check.yml]: Sure, updated to `pip install pylint==3.1.0`

**Pull request merged**